.github/workflows/release_nightly.yml

```diff
@@ -486,46 +452,102 @@ jobs:
+      attestations: write
+      checks: read
+      contents: write
+      id-token: write
+      pull-requests: read
+      statuses: write
+    steps:
+      - name: Clone Ruffle repo
         uses: actions/checkout@v4
+
+      - name: Install Rust toolchain
+        uses: dtolnay/rust-toolchain@stable
         with:
-          repository: ruffle-rs/demo
-          path: demo
-          ref: master
-          fetch-depth: 0
-          persist-credentials: false # Needed to allow commit via RUFFLE_BUILD_TOKEN below
+          toolchain: stable
+          targets: wasm32-unknown-unknown
+          components: rust-src
 
-      - name: Update web demo
-        if: matrix.demo
-        run: |
-          # Delete the old build.
-          rm -fr *
+      - name: Setup Node.js
+        uses: actions/setup-node@v4
+        with:
+          node-version: "22"
+          registry-url: https://registry.npmjs.org
 
-          # Copy the fresh build into this folder.
-          cp -fr ../web/packages/demo/dist/* .
+      # wasm-bindgen-cli version must match wasm-bindgen crate version.
+      # Be sure to update in test_web.yml, Cargo.toml, web/docker/Dockerfile,
+      # and web/README.md as well.
+      - name: Install wasm-bindgen
+        run: cargo install wasm-bindgen-cli --version 0.2.100
 
-          # Restore our custom swfs
-          git restore swfs swfs.json
+      # Keep the version number in sync in all workflows,
+      # and in the extension builder Dockerfile!
+      - name: Install wasm-opt
+        uses: sigoden/install-binary@v1
+        with:
+          repo: WebAssembly/binaryen
+          tag: version_122
+          name: wasm-opt
 
-          # Create git commit. Amend previous commit to avoid daily commit spam.
-          git config user.name "RuffleBuild"
-          git config user.email "ruffle@ruffle.rs"
-          git add -A
-          git commit --amend -m "Nightly build ${{ needs.create-nightly-release.outputs.date }}"
-        working-directory: demo
+      - name: Install node packages
+        working-directory: web
+        shell: bash -l {0}
+        run: npm ci
 
-      - name: Push web demo
-        if: github.repository == 'ruffle-rs/ruffle' && matrix.demo
-        uses: ad-m/github-push-action@master
-        with:
-          repository: ruffle-rs/demo
-          github_token: ${{ secrets.RUFFLE_BUILD_TOKEN }}
-          directory: demo
-          force: true
+      - name: Seal version data
+        shell: bash -l {0}
+        working-directory: web
+        env:
+          BUILD_ID: ${{ github.run_number }}
+          ENABLE_VERSION_SEAL: "true"
+          FIREFOX_EXTENSION_ID: ${{ secrets.FIREFOX_EXTENSION_ID }} # Needed to inject into manifest.json
+        run: npm run version-seal
+
+      - name: Build web
+        env:
+          BUILD_ID: ${{ github.run_number }}
+          # Build web demo with WebGPU support for testing in Chrome origin trial on ruffle.rs
+          CARGO_FEATURES: jpegxr,webgpu
```

Comment:
I know this builds off my changes which had the same change but before my PR or this PR webgpu was only built for the demo, now it's built for selfhosted too, is that a problem?

Reply:
As I wrote on Discord:

> On September 4, 2022 a toggle was added for webgpu and this was added: https://github.com/ruffle-rs/ruffle/pull/7777. However, on January 13, 2023, wgpu-webgl become the default: https://github.com/ruffle-rs/ruffle/pull/9134. wgpu-webgl uses both the feature wgpu uses and webgl through wgpu: https://github.com/ruffle-rs/ruffle/blob/a2c3d5e7dd54ff9a50de1f2c6bf9011237c3b73c/web/Cargo.toml#L29-L30 So is there still a reason to use the webgpu feature for the web demo?

Basically, unless I misunderstand I'm not sure we need the webgpu cargo feature at all in any workflow. Do we?

Reply:
> I think if we want to make the PR refactor-only and not have any major user-facing implications, then disabling webgpu is okay.
But not being able to deploy webgpu demo-only anymore is kinda unfortunate, if we ever wanted it.

@adrian17 on Discord

Reply:
Removed the webgpu feature for now, but added a note about this.